client/src/lib.rs

```diff
@@ -174,6 +174,30 @@ impl BpxClient {
         Ok(res)
     }
 
+    /// Deserializes the response body as JSON, logging the body at error level on failure.
+    ///
+    /// Unlike `Response::json()`, this method captures the response body and logs it
+    /// if deserialization fails, making it easier to debug API changes.
+    pub async fn json_with_context<T: serde::de::DeserializeOwned>(res: Response) -> Result<T> {
+        let body = res.text().await?;
+        serde_json::from_str(&body).map_err(|e| {
+            // Truncate very long responses for readability in logs
+            let body_preview = if body.len() > 2000 {
+                format!(
+                    "{}...[truncated, {} bytes total]",
+                    &body[..2000],
+                    body.len()
+                )
+            } else {
+                body
+            };
+            tracing::error!(
+                "Failed to deserialize API response: {e}\nResponse body: {body_preview}"
+            );
+            e.into()
```

Comment:
I think we typically use fields for logs, e.g.:

```suggestion
        serde_json::from_str(&body).map_err(|error| {
            // Truncate very long responses for readability in logs
            let body_preview = if body.len() > 2000 {
                format!(
                    "{}...[truncated, {} bytes total]",
                    &body[..2000],
                    body.len()
                )
            } else {
                body
            };
            tracing::error!(
                %error,
                body_preview, 
                "Failed to deserialize API response"
            );
            error.into()
```